src/site/topic24.rst

```diff
@@ -106,6 +106,43 @@ Add to Binary Search Tree
 Minimum & Maximum
 =================
 
+.. code-block:: java
+    :linenos:
+
+    public T min() {
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
+        return min(root);
+    }
+
+    private T min(Node<T> current) {
+        if (current.getLeft() == null) {
+            return current.getData();
+        } else {
+            return min(current.getLeft());
+        }
+    }
+
+
+    public T max() {
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
```

Comment:
Could also return `Optional<T>` which is a fancy new trend. Semi-functional-style-programming-sorta-dont-think-about-it-much.